### PR TITLE
Improving wind offshore reporting

### PR DESCRIPTION
**Pull request by cchrisgong**

### CODE REVIEW

File: R/compareScenarios.R [hunk 1]
```diff
@@ -738,11 +738,16 @@ compareScenarios <- function(mif, hist,
              "SE|Electricity|Biomass|w/o CCS (EJ/yr)",
              "SE|Electricity|Solar|CSP (EJ/yr)",
              "SE|Electricity|Solar|PV (EJ/yr)",
-             "SE|Electricity|Wind|Onshore (EJ/yr)",
-             "SE|Electricity|Wind|Offshore (EJ/yr)",
              "SE|Electricity|Hydrogen (EJ/yr)",
              "SE|Electricity|Net Imports (EJ/yr)")
 
+  if ("SE|Electricity|Wind|Offshore (EJ/yr)" %in% magclass::getNames(data, dim = 3)) {
+    items <- append(items, c( "SE|Electricity|Wind|Onshore (EJ/yr)",
```
Comment (pfuehrlich-pik) [inline R/compareScenarios.R:745]: The name "items" is quite general, is there a better name to describe whats actually in this variable?

File: R/reportCapacity.R [hunk 2]
```diff
@@ -93,10 +93,12 @@ reportCapacity <- function(gdx, regionSubsetList = NULL, t = c(seq(2005, 2060, 5
   tmp <- mbind(tmp, setNames(dimSums(vm_cap[, , "dot"], dim = 3),           "Cap|Electricity|Oil|w/o CCS (GW)"))
   tmp <- mbind(tmp, setNames(dimSums(vm_cap[, , "spv"], dim = 3),                 "Cap|Electricity|Solar|PV (GW)"))
   tmp <- mbind(tmp, setNames(dimSums(vm_cap[, , "csp"], dim = 3),                 "Cap|Electricity|Solar|CSP (GW)"))
-  tmp <- mbind(tmp, setNames(dimSums(vm_cap[, , "wind"], dim = 3),                "Cap|Electricity|Wind|Onshore (GW)"))
+  
   if ("windoff" %in% magclass::getNames(vm_cap, dim = 1)) {
-    tmp <- mbind(tmp, setNames(dimSums(vm_cap[, , "windoff"], dim = 3),                "Cap|Electricity|Wind|Offshore (GW)"))
-  }
+    tmp <- mbind(tmp, setNames(dimSums(vm_cap[, , "wind"], dim = 3),                "Cap|Electricity|Wind|Onshore (GW)"))
```
Comment (pfuehrlich-pik) [inline R/reportCapacity.R:98]: I'm not a fan of the name "tmp", as it does not describe what's stored in the variable at all.

File: R/reportCapacity.R [hunk 3]
```diff
@@ -139,9 +141,13 @@ reportCapacity <- function(gdx, regionSubsetList = NULL, t = c(seq(2005, 2060, 5
   tmp2 <- mbind(tmp2, setNames(dimSums(vm_deltaCap[, , "ngcc"], dim = 3),           "New Cap|Electricity|Gas|CC|w/o CCS (GW/yr)"))
   tmp2 <- mbind(tmp2, setNames(dimSums(vm_deltaCap[, , "spv"], dim = 3),            "New Cap|Electricity|Solar|PV (GW/yr)"))
   tmp2 <- mbind(tmp2, setNames(dimSums(vm_deltaCap[, , "csp"], dim = 3),            "New Cap|Electricity|Solar|CSP (GW/yr)"))
-  tmp2 <- mbind(tmp2, setNames(dimSums(vm_deltaCap[, , "wind"], dim = 3),           "New Cap|Electricity|Wind|Onshore (GW/yr)"))
+
   if ("windoff" %in% magclass::getNames(vm_deltaCap, dim = 1)) {
+    tmp2 <- mbind(tmp2, setNames(dimSums(vm_deltaCap[, , "wind"], dim = 3),           "New Cap|Electricity|Wind|Onshore (GW/yr)"))
```
Comment (pfuehrlich-pik) [inline R/reportCapacity.R:146]: Having multiple different basically unnamed variables makes the code even harder to read. I would suggest to replace all tmp* variables with more meaningful names.

**Review by pfuehrlich-pik — CHANGES_REQUESTED** [review]
Just some comments on variable naming. I'm not working with remind2, so I can't say anything about the actual changes.

**Review by fschreyer — COMMENTED** [review]
Looks good! Did you test compareScenarios with a mif without windoff? Wind now there again in the electricity mix?

File: R/reportCapacity.R (hunk 3 above)
Comment (fschreyer) [reply R/reportCapacity.R:146]: In principle, I agree with what you say, Pascal. Unfortunately, you will find things like things all through the reporting and other libraries. So probably not an issue to tackle immediately.

**Comment by cchrisgong** [comment]
> Looks good! Did you test compareScenarios with a mif without windoff? Wind now there again in the electricity mix?

yes, I tested them out, if you want to see them (for run with and without windoff), they are here (and nice looking ^^): /p/tmp/chengong/remind_offshore/remind/windoff_compareScen

**Pull request merged**